plugins/flytekit-greatexpectations/flytekitplugins/great_expectations/schema.py

```diff
@@ -0,0 +1,310 @@
+    This is because base directory which has the dataset file 'must' be given in GreatExpectations' config file
+    """
+    local_file_path: Optional[str] = None
+    checkpoint_params: Optional[Dict[str, Union[str, List[str]]]] = None
+    batch_request_config: BatchRequestConfig = None
+    context_root_dir: str = "./great_expectations"
+
+
+class GreatExpectationsType(object):
+    """
+    Use this class to send the GreatExpectationsFlyteConfig.
+
+    Args:
+        config: GreatExpectations Plugin configuration
+
+    TODO: Connect Data Docs to Flyte Console.
+    """
+
+    @classmethod
+    def config(cls) -> Tuple[Type, Type[GreatExpectationsFlyteConfig]]:
+        return (
+            str,
+            GreatExpectationsFlyteConfig(datasource_name="", data_connector_name="", expectation_suite_name=""),
+        )
+
+    def __class_getitem__(cls, config: Tuple[Type, GreatExpectationsFlyteConfig]) -> Any:
+        if not (isinstance(config, tuple) or len(config) != 2):
+            raise AssertionError("GreatExpectationsType must have both datatype and GreatExpectationsFlyteConfig")
+
+        class _GreatExpectationsTypeClass(GreatExpectationsType):
+            __origin__ = GreatExpectationsType
+
+            @classmethod
+            def config(cls) -> Tuple[Type, Type[GreatExpectationsFlyteConfig]]:
+                return config
+
+        return _GreatExpectationsTypeClass
+
+
+class GreatExpectationsTypeTransformer(TypeTransformer[GreatExpectationsType]):
+    def __init__(self):
+        super().__init__(name="GreatExpectations Transformer", t=GreatExpectationsType)
+
+    @staticmethod
+    def get_config(t: Type[GreatExpectationsType]) -> Tuple[Type, Type[GreatExpectationsFlyteConfig]]:
+        return t.config()
+
+    def get_literal_type(self, t: Type[GreatExpectationsType]) -> LiteralType:
+        datatype = GreatExpectationsTypeTransformer.get_config(t)[0]
+
+        if issubclass(datatype, str):
+            return LiteralType(simple=_type_models.SimpleType.STRING, metadata={})
+        elif issubclass(datatype, FlyteFile):
+            return FlyteFilePathTransformer().get_literal_type(datatype)
+        elif issubclass(datatype, FlyteSchema):
+            return FlyteSchemaTransformer().get_literal_type(datatype)
+
+    def to_literal(
+        self,
+        ctx: FlyteContext,
+        python_val: Union[FlyteFile, FlyteSchema, str],
+        python_type: Type[GreatExpectationsType],
+        expected: LiteralType,
+    ) -> Literal:
+        datatype = GreatExpectationsTypeTransformer.get_config(python_type)[0]
+
+        if issubclass(datatype, FlyteSchema):
+            return FlyteSchemaTransformer().to_literal(ctx, python_val, datatype, expected)
+        elif issubclass(datatype, FlyteFile):
+            return FlyteFilePathTransformer().to_literal(ctx, python_val, datatype, expected)
+
+        return Literal(scalar=Scalar(primitive=Primitive(string_value=python_val)))
```

Comment:
what about dictionaries or custom objects etc? I think we should raise an error if the type does not match?
Else we should forward the type lookup to the TypeEngine?

Reply:
Today we support string, FlyteFile, and FlyteSchema datatypes. I don't think any other datatype fits into the Great Expectations ecosystem. Thus, I have added type-checking in this block of code.
I believe there's type checking included everywhere else. Please let me know if that's not the case.